android/src/main/java/com/reactcommunity/rndatetimepicker/RNTimePickerDialogFragment.java

```diff
@@ -55,9 +56,18 @@ static TimePickerDialog getDialog(
     final int minute = date.minute();
     boolean is24hour = DateFormat.is24HourFormat(activityContext);
 
+    int minuteInterval = RNConstants.DEFAULT_TIME_PICKER_INTERVAL;
+    if (args != null && RNMinuteIntervals.isValid(args.getInt(RNConstants.ARG_INTERVAL, -1))) {
+      minuteInterval = args.getInt(RNConstants.ARG_INTERVAL);
+    }
+
     RNTimePickerDisplay display = RNTimePickerDisplay.DEFAULT;
     if (args != null && args.getString(RNConstants.ARG_DISPLAY, null) != null) {
-      display = RNTimePickerDisplay.valueOf(args.getString(RNConstants.ARG_DISPLAY).toUpperCase(Locale.US));
+      if (RNMinuteIntervals.isRadialPickerCompatible(minuteInterval)) {
+        display = RNTimePickerDisplay.valueOf(args.getString(RNConstants.ARG_DISPLAY).toUpperCase(Locale.US));
+      } else {
+        display = RNTimePickerDisplay.SPINNER;
```

Comment:
Why you are setting the display spinner? You can get from `args.getString(RNConstants.ARG_DISPLAY)`

Reply:
I'm not strong in java, I just fixed some issues from PR #66 to build this and get right necessary functionality in my project, according to the rule "works - don't touch it". In this case I don't understand what are you talking about. I'll be appreciated for more detailed explanation

Reply:
You are setting the `display` without respect the value of the `display` passed by the user is that right? I think you can check if the user passed the prop checking the `ARG_DISPLAY` if has value you se into `display` if not you set `SPINNER` what do you think?